netket/graph/abstract_graph.py

```diff
@@ -0,0 +1,51 @@
+import abc
+import numpy as _np
+
+
+class AbstractGraph(abc.ABC):
+    """Abstract class for NetKet graph objects"""
+
+    @property
+    @abc.abstractmethod
+    def is_connected(self):
+        r"""bool: True if the graph is connected"""
+        return NotImplementedError
+
+    @property
+    @abc.abstractmethod
+    def is_bipartite(self):
+        r"""bool: True if the graph is bipartite"""
+        return NotImplementedError
+
+    @property
+    @abc.abstractmethod
+    def edges(self):
+        r"""list: List containing the edges of the graph"""
+        return NotImplementedError
+
+    @property
+    @abc.abstractmethod
+    def distances(self):
```

Comment:
For the same reason we discussed regarding `automorphisms`, I think `distances` should not be a property but a method, so that it is more clear to callers that this can perform expensive computation.